ballerina/data-loaders.bal

```diff
@@ -14,9 +14,102 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/jballerina.java;
+import ballerina/file;
+
 # Represents a data loader that can load documents from various sources.
 public type DataLoader isolated object {
     # Loads documents from a source.
     # + return - document or an array of documents, or an `ai:Error` if the loading fails
     public isolated function load() returns Document[]|Document|Error;
 };
+
+# Dataloader that can be used to load supported file types as `TextDocument`s.
+# Currently only supports `pdf`, `docx` and `pptx` file types.
+public isolated class TextDataLoader {
+    *DataLoader;
+    final readonly & string[] paths;
+
+    public isolated function init(string ...paths) returns Error? {
+        // Check if the file exists by trying to get metadata
+        foreach string path in paths {
+            file:MetaData|error metadata = file:getMetaData(path);
+            if metadata is error {
+                return error Error("File does not exist: " + path);
+            }
+        }
+        self.paths = paths.cloneReadOnly();
+    }
+
+    public isolated function load() returns Document[]|Document|Error {
+        Document[] documents = from string path in self.paths select check loadDocument(path);
+        if documents.length() == 1 {
+            return documents[0];
+        }
+        return documents;
+    }
+}
+
+isolated function loadDocument(string path) returns Document|Error {
+        string? fileType = getFileType(path);
+        if fileType is () {
+            string extension = getFileExtension(path);
+            return error Error(string `Unsupported file type: ${extension}`);
+        }
+
+        match fileType {
+            "pdf" => {
+                TextDocument|error result = readPdfNative(path);
+                if result is error {
+                    return error Error(result.message());
+                }
+                return result;
+            }
+            "docx" => {
+                TextDocument|error result = readDocxNative(path);
+                if result is error {
+                    return error Error(result.message());
+                }
+                return result;
+            }
+            "pptx" => {
+                TextDocument|error result = readPptxNative(path);
+                if result is error {
+                    return error Error(result.message());
+                }
+                return result;
+            }
+        }
+        return error Error("Unexpected error in file type processing");
+}
+
+isolated function getFileType(string path) returns "pdf"|"docx"|"pptx"? {
```

Comment:
seems like these string literals are used in multiple places. Shall we create an enum for possible supported types ?